include/swift/SILOptimizer/Analysis/EscapeAnalysis.h

```diff
@@ -234,6 +246,18 @@ class EscapeAnalysis : public BottomUpIPAnalysis {
     Global
   };
 
+  // Must be ordered from most precise to least precise. A meet across memory
+  // locations (such as aggregate fields) always moves down.
+  enum PointerKind {
+    NoPointer,
+    NoNonReferencePointer,
+    MustHaveReference,
+    AnyPointer
+  };
+  static bool containsNoNonReference(PointerKind pointerKind) {
```

Comment:
maybe call this "canOnlyContainReferences"?

Reply:
Agreed